File: core/src/main/java/com/toocol/ssh/core/term/core/ActionCRLF.java
package com.toocol.ssh.core.term.core;

import com.toocol.ssh.utilities.event.CharEvent;
import com.toocol.ssh.utilities.utils.StrUtil;

/**
 * @author ZhaoZhe (joezane.cn@gmail.com)
 * @date 2022/4/25 18:06
 */
public final class ActionCRLF extends TermCharAction {
    @Override
    public CharEvent[] watch() {
        return new CharEvent[]{CharEvent.CR, CharEvent.LF};
    }

    @Override
    public boolean act(Term term, CharEvent charEvent, char inChar) {
        int[] cursorPosition = term.getCursorPosition();
        term.hideCursor();
        term.setCursorPosition(Term.getPromptLen(), cursorPosition[1]);
        term.showCursor();
        term.historyCmdHelper.push(term.lineBuilder.toString());

        term.executeCursorOldX.set(Term.getPromptLen());
        term.printExecution(StrUtil.EMPTY);
        return true;
    }
}

File: core/src/main/java/com/toocol/termio/core/term/core/ActionCRLF.kt
package com.toocol.termio.core.term.core

import com.toocol.termio.core.term.core.Term.Companion.promptLen
import com.toocol.termio.utilities.event.CharEvent
import com.toocol.termio.utilities.utils.StrUtil

/**
 * @author ZhaoZhe (joezane.cn@gmail.com)
 * @date 2022/4/25 18:06
 */
class ActionCRLF : TermCharAction() {
    override fun watch(): Array<CharEvent?> {
        return arrayOf(CharEvent.CR, CharEvent.LF)
    }

    override fun actOnConsole(term: Term, charEvent: CharEvent, inChar: Char): Boolean {
        val cursorPosition = term.cursorPosition
        term.hideCursor()
        term.setCursorPosition(promptLen, cursorPosition[1])
        term.showCursor()
        term.historyCmdHelper.push(term.lineBuilder.toString())
        term.executeCursorOldX.set(promptLen)
        term.printExecution(StrUtil.EMPTY)
        return true
    }

    override fun actOnDesktop(term: Term, charEvent: CharEvent, inChar: Char): Boolean {
        return true
    }
}